test/uploadInBackground.test.ts

```diff
@@ -0,0 +1,233 @@
+import { assert } from "chai";
+import { ImportMock } from "ts-mock-imports";
+import { UploadJobStatus, UrlToJobId } from "../src/types";
+import * as uploadInBackgroundFunctions from "../src/api/actions/uploadInBackground";
+import * as checkBulkUploadJob from "../src/api/actions/checkBulkUploadJob";
+import * as startBulkUpload from "../src/api/actions/startBulkUpload";
+
+describe("uploadInBackground", async () => {
+  const fakeApiUri = "";
+  const fakeUrl = "fakeJPEG.com";
+  let fakeUrlList: string[] = [];
+  let fakeStartJobResponse: UrlToJobId = {};
+  let fakeCheckJobResponse: UploadJobStatus = {};
+
+  beforeEach(() => {
+    fakeUrlList = [];
+    fakeStartJobResponse = {};
+    fakeCheckJobResponse = {};
+  });
+
+  it("Calls startBulkUpload in batches", () => {
+    for (let i = 0; i < 101; i++) {
+      fakeUrlList[i] = fakeUrl + i;
+      fakeStartJobResponse[fakeUrlList[i]] = i.toString();
+    }
+    const mockUpload = ImportMock.mockFunction(
+      uploadInBackgroundFunctions,
+      "tryStartBulkUpload",
+      fakeStartJobResponse
+    ).resolves(fakeStartJobResponse);
```

Comment:
why do we `resolves` ?

Reply:
`resolves` creates a promise which evaluates to the argument given - `tryStartBulkUpload` returns a promise, so our mock should return a promise as well